keps/sig-apps/2255-pod-cost/README.md

```diff
@@ -55,163 +54,183 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
+as to which pods should be deleted first on scale down.
 
 ## Motivation
 
-For some applications, it is necessary that the application can tell Kubernetes which pod can be deleted and which replica has to be protected. The reason for this is that some applications do have stateful sessions and it is not possible to put such an application into Kubernetes because of session termination resulting from "random" down-scale. If the application is able to tell Kubernetes which of the replicas contains no/few/less important active sessions, this would solve many problems. This feature is non-disruptive to the default behaviour. Only if the annotation is existing, it will make a difference in deletion order.
+Currently ReplicaSets are scaled down based on a criteria that on the
+limit prioritizes deleting pods with a more recent creation/readiness 
+timestamp. This is not ideal for some applications where the cost of 
+deleting pods is not related to how recent they were created. 
 
 ### Goals
 
-To recommend which pod gets deleted next of a ReplicaSet. This should help to avoid major reworks in existing applications architecture:
-* [45509](https://github.com/kubernetes/kubernetes/issues/45509) - Scale down a deployment by removing specific pods
+- An API that allows applications to influence the order of deleting 
+  pods when scaling down a ReplicaSet
 
 
 ### Non-Goals
 
-Guaranteed (in contrast to the recommendation stated in Goals) deletion of a selected replica.
+- Guarantees on pod deletion order
+- A controller that sets the cost of deleting the pods.
 
 ## Proposal
 
-The application can set the `controller.kubernetes.io/pod-cost` annotation to a pod through the Kubernetes API. When a downscale event happens, the pod with the lower priority value of the previously set annotation will be deleted first. If one pod of the Deployment has no priority annotation set, it will be treated as the lowest priority.
-
-If all pods have the same priority, there is no difference in the normal pod delete decision behaviour. The same applies if the pod-cost annotation is not used at all.
-
-The pod-cost annotation can be changed during operation, for example, if workload changes or a new master gets elected.
+Define a known annotation, namely `controller.kubernetes.io/pod-deletion-cost` that 
+applications can set to offer a hint on the cost of deleting a pod compared
+to other pods belonging to the same ReplicaSet. 
 
 ### User Stories (optional)
 
-
 #### Story 1
 
-In an application environment with stateful worker (user-)sessions, it is essential to keep the user sessions alive as good as possible. In case of a scale-down event, the application has to tell the scheduler, which delete decision would have the lowest impact on existing sessions.
+The different pods of an application could have different utilization levels. 
+On scale down, the application may prefer to remove the pods with lower utilization.
```

Comment:
I'm not convinced about utilization example. The reason is that utilization is often changing very frequently and  we don't want to have gazillions of pod updates frequently updating that annotation.

Reply:
> To avoid frequently updating the pods, the application should update pod-deletion-cost
once before issuing a scale down.

But it's not always application that is issue'ing scale-down, right? It may be HPA for example...

Reply:
right, but that doesn't mean the story is not valid, there are cases where the application is in control, which is what this story is emphasizing.

Reply:
OK - let's make it more explicit that it's the application itself that is in control of scale downs in this particular example.

Reply:
done, and also gave an example.

Reply:
Thanks!